documentation/docs/10-adapters.md

```diff
@@ -56,6 +57,67 @@ You can also use `adapter-static` to generate single-page apps (SPAs) by specify
 
 > You must ensure [`trailingSlash`](configuration#trailingslash) is set appropriately for your environment. If your host does not render `/a.html` upon receiving a request for `/a` then you will need to set `trailingSlash: 'always'` to create `/a/index.html` instead.
 
+#### GitHub Pages
+
+SvelteKit can be automatically configured for deployment on GitHub Pages by using the [`actions/configure-pages`](https://github.com/actions/configure-pages) GitHub Action before building it. This action will set `GITHUB_PAGES=true` and replace `kit.paths.base` in `svelte.config.js` by the correct path for your GitHub repository.
```

Comment:
the automatic `kit.paths.base` replacement makes me super nervous. hacks like this are guaranteed to break for a percentage of users, and it's hard to test locally. it should really be an environment variable that you add to your config — does something like this exist?

```js
// svelte.config.js
export default {
  kit: {
    adapter: adapter(),
    paths: { base: process.env.GITHUB_PAGES_PATH ?? '' }
  }
};
```